contracts/modules/cover/Cover.sol

```diff
@@ -346,43 +438,45 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
 
         // check if this request should be skipped, keeping the previous allocation
         if (poolAllocationRequests[i].skip) {
-          coverSegmentAllocations[allocationRequest.coverId][segmentId].push(previousPoolAllocation);
-          totalCoverAmountInNXM += previousPoolAllocation.coverAmountInNXM;
+          coverSegmentAllocations[allocationRequest.coverId][params.segmentId].push(previousPoolAllocation);
+          vars.totalCoverAmountInNXM += previousPoolAllocation.coverAmountInNXM;
           continue;
         }
 
-        vars.previousPremiumInNXM = previousPoolAllocation.premiumInNXM;
-        vars.refund =
-          previousPoolAllocation.premiumInNXM
-          * (allocationRequest.previousExpiration - block.timestamp) // remaining period
-          / (allocationRequest.previousExpiration - allocationRequest.previousStart); // previous period
-
         // get stored allocation id
         allocationRequest.allocationId = previousPoolAllocation.allocationId;
+
+        uint poolAllocationRatio =
+          previousPoolAllocation.coverAmountInNXM
+          * COVER_ALLOCATION_RATIO_DENOMINATOR
+          / vars.previousTotalCoverAmountInNXM;
+
+        vars.previousAllocationAmountInNXMRepriced =
+          poolAllocationRatio * oldSegmentAmountInNXMRepriced
+          / COVER_ALLOCATION_RATIO_DENOMINATOR;
+
       } else {
         // request new allocation id
         allocationRequest.allocationId = 0;
 
-        // zero out previous premium or refund
-        vars.previousPremiumInNXM = 0;
-        vars.refund = 0;
+        vars.previousAllocationAmountInNXMRepriced = 0;
```

Comment:
more broadly on naming:
- all allocations are in NXM always, so probably worth dropping "InNXM"
- all allocations are either pool allocations, or total allocation - so worth dropping "pool" everywhere or keep it everywhere - it's quite inconsistent atm